app/controllers/catalog_controller.rb

```diff
@@ -345,24 +342,26 @@ def load_bulk_jobs(druid)
 
   # Determines whether or not the current user has permissions to view the current DOR object.
   def valid_user?(dor_object)
-    begin
-      @apo = dor_object.admin_policy_object
-    rescue
-      return false
-    end
-
+    # First check the workgroups defined for admin, manager or viewer; see
+    # User::ADMIN_GROUPS, User::MANAGER_GROUPS and User::VIEWER_GROUPS
+    return true if @user.is_admin || @user.is_manager || @user.is_viewer
+    # If that fails, check an APO for permitted roles of the current_user.
+    valid = false
+    @apo = dor_object.admin_policy_object rescue nil
     if @apo
-      unless @user.is_admin || @user.is_viewer || dor_object.can_view_metadata?(@user.roles(@apo.pid))
-        render :status => :forbidden, :text => 'forbidden'
-        return false
-      end
+      valid = dor_object.can_view_metadata?(@user.roles(@apo.pid))
+      msg = 'Governing APO forbids access' unless valid
+    elsif dor_object.is_a? Dor::AdminPolicyObject
+      # In the absence of a governing APO, allow an APO to authorize itself.
+      valid = dor_object.can_view_metadata?(@user.roles(dor_object.pid))
+      msg = 'Item is an APO that forbids access' unless valid
     else
-      unless @user.is_admin || @user.is_viewer
-        render :status => :forbidden, :text => 'No APO, no access'
-        return false
-      end
+      msg = 'Item has no APO that allows access'
+    end
+    unless valid
+      render :status => :forbidden, :text => msg
     end
-    true
+    valid
   end
 
   def get_leafdir(directory)
```

Comment:
alright, i just spent a bit trying to figure out a way to reimplement `valid_user?` with `can_view_object?`.  here's what i came up with:

* though the two methods are effectively the same since they both use the same role list, `valid_user?` calls `can_view_metadata?`, but `can_view_object?` calls `can_view_content?`.  i have a vague memory of @darrenleeweber and @LynnMcRae discussing this recently?  anyway, if it'd be ok to have `valid_user?` rely on `can_view_content?` for its logic instead, i think the below code would be a more concise reimplementation with the same messages and logic and less duplication of the basic permission checking logic (though an assumption about the cause of access denial is made for message construction purposes):
```ruby
def valid_user?(dor_object)
  valid = @user.can_view_object? dor_object
  return valid if valid

  if dor_object.is_a? Dor::AdminPolicyObject
    msg = 'Item is an APO that forbids access'
  elsif dor_object.admin_policy_object
    msg = 'Governing APO forbids access'
  else
    msg = 'Item has no APO that allows access'
  end
  render :status => :forbidden, :text => msg
end
```